src/main_app.rs

```diff
@@ -0,0 +1,79 @@
+use crossbeam_channel::{select, unbounded, Receiver};
```

Comment:
What do you think about moving that to a `gui` folder?

Reply:
Or maybe create a `bin` folder, and the GUI App keeps in another directory.

Anyway, just some thoughts :)

Reply:
Agree. But let's first see how the file structure will go and separate in folders if we see clear chance for that.